src/Random/InverseTransformSampling.h

```diff
@@ -46,36 +46,161 @@ namespace ippl {
         template <class RegionLayout>
         InverseTransformSampling(Distribution &dist_r, Vector<T, Dim> &rmax_r, Vector<T, Dim> &rmin_r, RegionLayout& rlayout_r, size_type &ntotal_r)
         : dist_m(dist_r),
-        rmax_m(rmax_r),
-        rmin_m(rmin_r),
         ntotal_m(ntotal_r){
-            const typename RegionLayout::host_mirror_type regions = rlayout_r.gethLocalRegions();
-            
+
+            updateBounds(rmax_r, rmin_r, rlayout_r);
+
+        }
+
+        /*!
+         * @brief Constructor for InverseTransformSampling class without applying domain decomposition..
+         *
+         * @param dist_ The distribution to sample from.
+         * @param rmax_ Maximum range for sampling.
+         * @param rmin_ Minimum range for sampling.
+         * @param rlayout The region layout.
+         * @param ntotal_ Total number of samples to generate.
+        */
+	template <class RegionLayout>
+        InverseTransformSampling(Distribution &dist_r, Vector<T, Dim> &rmax_r, Vector<T, Dim> &rmin_r, Vector<T, Dim> &locrmax_r, Vector<T, Dim> &locrmin_r, size_type &ntotal_r)
+        : dist_m(dist_r),
```

Comment:
Can this constructor itself be used for the one with rlayout also so that you specify the local minimum and maximum from rlayout from the calling side? This way we can reduce the number of constructors

Reply:
I'm not following.
- What is the problem with having more constructors?
- if we remove the constructor with rlayout as argument, we have to change the constructor calls in all the alpine examples.

Reply:
I see. Checkout 07465ec. I tried to reduce some of the redundancies.

Reply:
Does your test still goes through?

Reply:
yes